src/feature/dirauth/dirvote.c

```diff
@@ -4539,6 +4569,10 @@ dirserv_generate_networkstatus_vote_obj(crypto_pk_t *private_key,
     bw_file_headers = smartlist_new();
     dirserv_read_measured_bandwidths(options->V3BandwidthsFile,
                                      routerstatuses, bw_file_headers);
+    /* Only set bw_file_digest255 when V3BandwidthsFile is configured. */
+    bwauth_bw_file_digest255(bw_file_digest255, options->V3BandwidthsFile,
```

Comment:
There is a possible race condition here if the file changes between reading it the first time and reading it the second time.  Maybe we could make the file only get read once?

Reply:
hmm, then it should be `dirserv_read_measured_bandwidths` that return the digest too. Does it sounds good?.
`dirserv_read_measured_bandwidths` uses `tor_getline`, so i could use something like strcpy and strcat to calculate the digest of all content?.
Or maybe we would like to change `dirserv_read_measured_bandwidths` to use `read_file_to_str` and then parse the lines with something like strtok?.
I'm not familiar the preferred functions to manipulate strings (and too used to easy python).

Reply:
in `dirserv_read_measured_bandwidths`?

> This pull request fails the Appveyor CI with some OpenSSL errors. Perhaps the merge base is wrong?
> 
> https://ci.appveyor.com/project/torproject/tor/builds/20235830/job/s9dqnp4mopk66q8c#L1289

Yes, i created the branch when appveyor was failing in master.
When we're ready to squash, i can rebase it to the commit that fixed appveyor or current master.

Reply:
We'll need to make sure that CI works after you rebase.